pkg/dashboard/resource/v3iostream.go

```diff
@@ -113,6 +199,127 @@ func (vsr *v3ioStreamResource) getNamespaceFromRequest(request *http.Request) st
 	return vsr.getNamespaceOrDefault(request.Header.Get("x-nuclio-project-namespace"))
 }
 
+func (vsr *v3ioStreamResource) getShardLagsMap(ctx context.Context, info v3ioStreamInfo) (map[string]restful.Attributes, error) {
+
+	// create v3io context
+	v3ioContext, err := v3iohttp.NewContext(vsr.Logger, &v3iohttp.NewContextInput{
+
+		// by default client id nil
+		HTTPClient: vsr.V3ioHTTPClient,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed creating v3io context")
+	}
+
+	// a data plane access key is required for v3io operations
+	accessKey := vsr.getCtxSession(ctx).GetPassword()
+	if accessKey == "" {
+		return nil, errors.Wrap(err, "A data plane access key is required")
+	}
+
+	dataPlaneInput := v3io.DataPlaneInput{
+		URL:           vsr.getPlatform().GetConfig().StreamMonitoring.WebapiURL,
+		AccessKey:     accessKey,
+		ContainerName: info.ContainerName,
+	}
+	getContainerContentsInput := &v3io.GetContainerContentsInput{
+		Path:             info.StreamPath,
+		GetAllAttributes: true,
+		DirectoriesOnly:  false,
+		DataPlaneInput:   dataPlaneInput,
+	}
+
+	shardLagsMap := map[string]restful.Attributes{}
+	lock := sync.Mutex{}
+
+	for {
+		response, err := v3ioContext.GetContainerContentsSync(getContainerContentsInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to get container subdirectories")
+		}
+
+		getContainerContentsOutput := response.Output.(*v3io.GetContainerContentsOutput)
+		response.Release()
+
+		errGroup, _ := errgroup.WithContextSemaphore(ctx, vsr.Logger, 10)
```

Comment:
perhaps that 10 needs to be configurable. can be on the streamMonitoring config with `GetStreamShardsConcurrenctRequests` or something equivalent. maybe on slower systems we would reduce it